sydent/http/matrixfederationagent.py

```diff
@@ -184,7 +184,7 @@ def endpointForURI(_uri):
 
     async def _route_matrix_uri(
         self, parsed_uri: "URI", lookup_well_known: bool = True
-    ):
+    ) -> "_RoutingResult":
```

Comment:
Why the quotes, out of curiosity?

Reply:
Python throws a name error on `_RoutingResult` without quotes even though the class is defined in the same file. So I added quotes, as this makes the interpreter and mypy happy. Let me know if it was the wrong thing to do and I'll be happy to fix it!

Reply:
Since `_RoutingResult` is further down in the file the parser doesn't yet know it is a symbol when parsing this class, putting it in quotes is the workaround. 👍

Reply:
Oh cool thanks for explaining that! Makes total sense.